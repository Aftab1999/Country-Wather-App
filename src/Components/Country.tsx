import React, { FC, useState } from "react";
import { Container, Box, Typography, TextField, Button } from "@mui/material";
import axios from "axios";
import { useNavigate } from "react-router-dom";

const Country: React.FC = () => {
  const navigate = useNavigate();
  // const [id, setId] = React.useState<IState["id"]>("")
  // const [id, setId] = React.useState('');

  const [data, setData] = useState<any>();
  const [input, setInput] = useState<string>("");

  const handleChange = async () => {
    const response = await axios.get(
      `https://restcountries.com/v3.1/name/${input}`
    );
    setData(response);
    console.log(response.data);
    navigate("/Details", { state: { data: response.data } });
  };

//   const handleDisable = (event) =>{
//     setId(event.target.value);
//     console.log(id)
// };

  return (
    <Container
      maxWidth="xs"
      sx={{
        marginTop: "100px",
        height: "300px",
        border: "1px solid black",
        bgcolor: " #e2e4e9;",
      }}
    >
      <Box
        style={{
          display: "flex",
          flexDirection: "column",
          alignItems: "center",
          marginTop: "50px",
        }}
      >
        <Typography variant="h4" data-testid ='Country_Data'>Country Data</Typography>
        <TextField
          style={{ marginBottom: "10px" }}
          id="textfield"
          label="Entry Country"
          onChange={(e) => setInput(e.target.value)}
         
        />
        <Button
          style={{ padding: 10, paddingRight: "80px", paddingLeft: "80px" }}
          variant="contained"
          id="submit"
          data-testid="submit"
          title="Country"
          onClick={handleChange}
         
        >
          Submit
        </Button>
      </Box>
    </Container>
  );
};
export default Country;
