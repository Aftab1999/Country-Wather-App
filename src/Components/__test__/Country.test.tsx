// import React from "react";
import { render, screen, fireEvent, act } from "@testing-library/react";
import Country from "../Country"

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom') as any,
    useNavigate: () => ({ navigate: jest.fn() }),
   useLocation: () => ({  }),
  
  }));

  // jest.mock("axios", () => ({
  //   get: jest.fn((_url, _body) => {
  //     return new Promise((resolve, reject) => {
  //       if (
  //         _url ===
  //         "https://restcountries.com/v3.1/name/india"
  //       ) {
  //         reject(new Error("Id not found!"));
  //       } else {
  //         resolve({
  //           data: {
  //             near_earth_objects: [{ id: "000000" }],
  //             id: "123",
  //           },
  //           status: 200,
  //         });
  //       }
  //     });
  //   }),
  // }));

  test("checking snapshot", () => {
    const test1 = render(<Country/>);
    expect(test1).toMatchSnapshot();
  });

  test("checking without crashes", () => {
    const test2 = render(<Country/>);
    expect(test2).toBeTruthy();
  });

  it ('Components exists or not', async() =>{
    const root= render(<Country/>)
    expect(root).toBeTruthy();
       
    })

it ("Check Rander Country", ()=>{
    const {queryByTitle} = render(<Country/>);
    const btn:any = queryByTitle("Country");
    expect(btn).toBeTruthy();
}) 

it ('checkSubmitButton', async() =>{
    render(<Country/>)
    expect(screen).toMatchSnapshot();
})

it ('checkSubmitButton',async() =>{
    render(<Country/>);
    const headingElement = await screen.getByTestId('submit')
    expect(headingElement).toBeInTheDocument();
  });

//   it('check button disable',async() =>{
//     const{getByTestId} =render(<Country/>);
//     const submit =getByTestId('submit');
//      expect(submit).toBeDisabled();
//    });

it('checkSubmitButton',async() =>{
    render(<Country/>);
    const headingElement = await screen.getByText('Submit')
    expect(headingElement).toBeInTheDocument();
  });

  describe ('Initial test, validate fields', () => {
    test('TextField component should exists.', () => {
      render(<Country/>);
      expect('Textfield').toBeDefined();
    });
  });

  it("checking for text content", async() => {
    const { getByTestId } = render(<Country/>)
    const text = getByTestId("Country_Data")
    expect(text.textContent).toBe("Country Data")
    
    })

    test("checking submitbutton", async () => {
      const { getByTestId } = render(<Country/>);
      const submitButton = getByTestId("submit");
      await act(async () => {
        await fireEvent.click(submitButton);
      });
      expect(submitButton).toBeDefined();
    });







// describe('render a TextField', () =>{
// const { getByText } = render(<Country/>);
// const result = find('#textfield').length;
// expect(result).toEqual(1);
// }) 