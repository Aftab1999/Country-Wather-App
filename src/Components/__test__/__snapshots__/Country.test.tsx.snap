// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkSubmitButton 1`] = `
Object {
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "logTestingPlaygroundURL": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports[`checking snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthXs css-1oizz9j-MuiContainer-root"
      >
        <div
          class="MuiBox-root css-0"
          style="display: flex; flex-direction: column; align-items: center; margin-top: 50px;"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
            data-testid="Country_Data"
          >
            Country Data
          </h4>
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            style="margin-bottom: 10px;"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="textfield"
              id="textfield-label"
            >
              Entry Country
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="textfield"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Entry Country
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="submit"
            id="submit"
            style="padding: 10px 80px 10px 80px;"
            tabindex="0"
            title="Country"
            type="button"
          >
            Submit
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthXs css-1oizz9j-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-0"
        style="display: flex; flex-direction: column; align-items: center; margin-top: 50px;"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
          data-testid="Country_Data"
        >
          Country Data
        </h4>
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          style="margin-bottom: 10px;"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="textfield"
            id="textfield-label"
          >
            Entry Country
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="textfield"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  Entry Country
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-testid="submit"
          id="submit"
          style="padding: 10px 80px 10px 80px;"
          tabindex="0"
          title="Country"
          type="button"
        >
          Submit
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
